### ISSUE TITLE
Android Obsidian 1.4.0, the language specific voices of Text to Speech can not select voice

### ISSUE THREAD

**Issue by ajmz** [joethei/obsidian-tts#21]
Android Obsidian 1.4.0, the language specific voices of Text to Speech can not select voice, while the Obsidian 1.0 for windows works well.

**Comment by joethei** [comment]
This plugin does not work at all on Android.
See the README and #8

**Comment by ajmz** [comment]
> This plugin does not work at all on Android. See the README and #8

Thanks a lot. I will report it to the development team of Obsidian to see if it can support TTS for Android environment.

**Comment by joethei** [comment]
No, we can't
(I am also part of that team)
The issue is with Android itself, see the link the README.

**Comment by ajmz** [comment]
Thanks for replying, I'm not a programmer.
But I read the README and the link inside to https://bugs.chromium.org/p/chromium/issues/detail?id=487255 .

I found the comment as follows, "But the problem I had was solvable":

[Comment 62](https://bugs.chromium.org/p/chromium/issues/detail?id=487255#c62) by [ubers...@gmail.com](https://bugs.chromium.org/u/3082663717) on Mon, Jun 21, 2021, 6:54 AM GMT+8
I might have misunderstood the purpose of this issue. But the problem I had was solvable.

I was using `utterance.voice` which is only wrong on Android Chrome. The following snippet is broken.

```
let utterance = new SpeechSynthesisUtterance("hello");
let voice = speechSynthesis.getVoices()[0]
utterance.voice = voice;
speechSynthesis.speak(utterance);
```

That works on Chrome Windows, iOS, MacOS, Firefox Android, but doesn't work on Android Chrome. The following works on all platforms I tested. To test in the console, you might have to run it twice to make sure the voices are loaded:

```
let utterance = new SpeechSynthesisUtterance("hello");
let voice = speechSynthesis.getVoices()[0]
utterance.voice = voice; // required for iOS
utterance.lang = voice.lang; // required for Android Chrome
utterance.voiceURI = voice.voiceURI; // unclear if needed
speechSynthesis.speak(utterance);
```

And you should hear a non-native accent saying "hello".

**Comment by usernotnull** [comment]
> ```
> let utterance = new SpeechSynthesisUtterance("hello");
> let voice = speechSynthesis.getVoices()[0]
> utterance.voice = voice; // required for iOS
> utterance.lang = voice.lang; // required for Android Chrome
> utterance.voiceURI = voice.voiceURI; // unclear if needed
> speechSynthesis.speak(utterance);
> ```
> 
> And you should hear a non-native accent saying "hello".

BTW, I tried the quoted workaround, but it didn't resolve the issue.
It's a shame, it would really benefit me to listen to my vault while on the go 😞

**Comment by ctkenyon** [comment]
Rather than waiting on Google to fix webview on Android (9+ years & counting) would it make sense to consider an alternate approach whereby the note could be shared with other apps capable of rendering markdown with built-in TTS facilities such as MS-EDGE? 

If That approach would be considered out of scope for this plug-in, are there others that I overlooked capable of doing that or worth requesting a plug-in specifically to share notes with other apps with specific features?
